pkg/controller/disruption/disruption.go

```diff
@@ -234,8 +234,13 @@ func NewDisruptionControllerInternal(
 // way to take advantage of listers with scale subresources, we use the workload
 // resources directly and only fall back to the scale subresource when needed.
 func (dc *DisruptionController) finders() []podControllerFinder {
-	return []podControllerFinder{dc.getPodReplicationController, dc.getPodDeployment, dc.getPodReplicaSet,
-		dc.getPodStatefulSet, dc.getScaleController}
+	return []podControllerFinder{
+		dc.getPodReplicationController,
+		dc.getPodDeployment,
+		dc.getPodReplicaSet,
+		dc.getPodStatefulSet,
+		dc.getScaleController,
+	}
```

Comment:
curious why this was reformatted but not changed... is there a reason not to implement a daemonset pod finder instead of using scale?

Reply:
@smarterclayton was originally for using the scale subresource IIUC: https://github.com/kubernetes/kubernetes/pull/98307#discussion_r568004947

I can see two reasons to have the scale subresource

- to not add a dependency on the Deployment controller in favor of the generic approach
- the scale subresource can be used by third party controllers to either achieve disruption or other goals

and one against

- there might be performance considerations of using the scale endpoint vs  the DaemonSet lister.